### PR TITLE
Extending ERC1155 with approvals by amount

### PR DESCRIPTION
**Pull request by ivanmmurciaua**
Approval by amount extension

### CODE REVIEW

File: EIPS/eip-approval_by_amount.md
Comment (MicahZoltu) [inline EIPS/eip-approval_by_amount.md:109]:
```suggestion
```
External links not allowed. You can include the test cases inline or as a file in `../assets/eip-5216/` and link to it here.  Alternatively, this is an optional section so you can leave it out for now.

File: EIPS/eip-approval_by_amount.md
Comment (Pandapip1) [inline EIPS/eip-approval_by_amount.md:11]:
```suggestion
requires: 20, 1155
```

EIP-20 is referenced

File: EIPS/eip-approval_by_amount.md
Comment (ivanmmurciaua) [reply EIPS/eip-approval_by_amount.md:4]: ERC or EIP?. I used EIP-4400 as a template for my proposal, and he titled ERC-721Consumable Extension.

File: EIPS/eip-approval_by_amount.md
Comment (ivanmmurciaua) [reply EIPS/eip-approval_by_amount.md:109]: I will make a new commit with assets for your revision. Thanks!

File: EIPS/eip-approval_by_amount.md
Comment (ivanmmurciaua) [reply EIPS/eip-approval_by_amount.md:11]: Name patterns are a require? I don't require ERC-20 for this extension

File: EIPS/eip-approval_by_amount.md
Comment (Pandapip1) [inline EIPS/eip-approval_by_amount.md:2]:
```suggestion
eip: 5216
```

File: EIPS/eip-approval_by_amount.md
Comment (lightclient) [reply EIPS/eip-approval_by_amount.md:4]: image

**Review by lightclient — CHANGES_REQUESTED** [review]
Can you also pls delete the `package-lock.json`?

File: EIPS/eip-approval_by_amount.md
Comment (ivanmmurciaua) [reply EIPS/eip-approval_by_amount.md]: Changed.

File: EIPS/eip-approval_by_amount.md
Comment (ivanmmurciaua) [reply EIPS/eip-approval_by_amount.md]: Relative links added

File: EIPS/eip-approval_by_amount.md
Comment (ivanmmurciaua) [reply EIPS/eip-approval_by_amount.md]: Relative links added

File: EIPS/eip-approval_by_amount.md
Comment (ivanmmurciaua) [reply EIPS/eip-approval_by_amount.md]: Relative links added

File: EIPS/eip-approval_by_amount.md
Comment (ivanmmurciaua) [reply EIPS/eip-approval_by_amount.md]: Renamed and changed

File: EIPS/eip-approval_by_amount.md
Comment (ivanmmurciaua) [reply EIPS/eip-approval_by_amount.md]: Relative links added

File: EIPS/eip-approval_by_amount.md
Comment (ivanmmurciaua) [reply EIPS/eip-approval_by_amount.md]: Relative links added

File: EIPS/eip-approval_by_amount.md
Comment (ivanmmurciaua) [reply EIPS/eip-approval_by_amount.md]: Removed

**Comment by ivanmmurciaua** [comment]
> Can you also pls delete the `package-lock.json`?

LICENSE and package-lock.json deleted

File: EIPS/eip-5216.md
Comment (SamWilsn) [inline EIPS/eip-5216.md:5]: This doesn't follow the format defined in [EIP-1](https://eips.ethereum.org/EIPS/eip-1#eip-header-preamble).

We have a linting tool that'll pick up on common mistakes, and you can trigger it by closing and re-opening your PR, if you'd like.

File: EIPS/eip-5216.md
Comment (SamWilsn) [inline EIPS/eip-5216.md:46]: You'll want to include 165 in your requires header.

File: EIPS/eip-5216.md
Comment (SamWilsn) [inline EIPS/eip-5216.md:116]:
```suggestion
Copyright and related rights waived via [CC0](../LICENSE.md).
```

File: EIPS/eip-5216.md
Comment (ivanmmurciaua) [reply EIPS/eip-5216.md:5]: Hi Sam, thanks for your review. 

Do you mean the format of the authors? I don't see discrepancies in eip, title and description format.

Comment (MicahZoltu) [reply EIPS/eip-5216.md:5]: @SamWilsn Lies.  😬  It appears eipw failed to validate the author line for some reason.

@ivanmmurciaua See the author line in https://raw.githubusercontent.com/ethereum/EIPs/master/eip-template.md for description and example of what a proper authors line should look like.  Email or GitHub handles only (GitHub preferred for notification reasons and bot functionality).

File: EIPS/eip-5216.md
Comment (ivanmmurciaua) [reply EIPS/eip-5216.md]: Changed!

File: EIPS/eip-5216.md
Comment (ivanmmurciaua) [reply EIPS/eip-5216.md:46]: Included!

File: EIPS/eip-5216.md
Comment (ivanmmurciaua) [reply EIPS/eip-5216.md]: So, yes. I had a discrepancy with author line but is already fixed. Thanks both @SamWilsn @MicahZoltu

File: assets/eip-5216/README.md
Comment (MicahZoltu) [inline assets/eip-5216/README.md:5]: Everything in this repository must be CC0 licensed.

**Review by MicahZoltu — COMMENTED** [review]
The goal of the reference implementation is to give readers a rough idea of what an implementation of this token might look like in as little code as possible so it remains relevant to the standard at hand and isn't bogged down with supplemental stuff.  This means that you should not include a complete project with dependencies, a build system, test infrastructure, etc. as that sort of content is much better kept in some repository that is actively maintained.

In this case, I recommend removing everything in `assets/eip-5216` except `ERC1155ApprovalByAmount.sol`, `IERC1155ApprovalByAmount`, and *maybe* `ExampleToken.sol`.  I would also combine them all into one file rather than spreading them out over several files, and in place of the dependency on `@openzeppelin/*` just put `ERC1155.sol`/`IERC1155.sol` and leave it as an exercise for the reader to find/implement ERC-1155.  You can also completely remove the `Ownable.sol` reference as that is out of scope of this standard.

For the test cases, it is valuable to include test cases but they should ideally be in a form that is somewhat platform agnostic.  An ideal set of test cases is a JSON file or something that is an array of `function, inputs, expected_outputs` or similar.  You may want to include some mechanism for representing postcondition checking and/or event checking as well if appropriate.  What isn't necessary is to have a fully executable test suite written in a particular language and doing so can make it harder for implementers to adapt to their own infrastructure (which may be wildly different from yours).

File: assets/eip-5216/README.md
Comment (ivanmmurciaua) [reply assets/eip-5216/README.md:5]: I'll delete everything except a .sol with reference implementation.

Also I will delete test cases. I'll maintain a repository with a complete project with deps, a build system and test infrastructure in my personal account.

Comment (MicahZoltu) [reply assets/eip-5216/README.md]: Looks much better now, thanks!  Also Note: Test Cases and Reference Implementation are both optional for ERCs, so if in the future you are struggling to come up with a trimmed down set of test cases or a trimmed down reference implementation you *can* leave them out entirely.  Generally it is better to include them if at all possible.

Comment (ivanmmurciaua) [reply assets/eip-5216/README.md]: Hello @MicahZoltu, thank you for everything.

Of course I will do it. In the .sol I left the example token as a reference implementation, but for the test cases, I would need an example to look at to make my test cases according to the style of this magnificent repository.

So, do you think it can advance to a next step this eip?

Comment (MicahZoltu) [reply assets/eip-5216/README.md]: I'm not an ERC reviewer, so you'll have to wait for one of the people the bot mentioned to review this before it can get merged.  I just try to help out from time to time by providing feedback to authors so they don't get held up unnecessarily.

Comment (ivanmmurciaua) [reply assets/eip-5216/README.md]: Thank you very much then @MicahZoltu. Appreciated 😁

File: EIPS/eip-5216.md
Comment (Pandapip1) [inline EIPS/eip-5216.md:100]:
```suggestion
This standard is compatible with [EIP-1155](./eip-1155.md).
```

File: assets/eip-5216/ERC1155ApprovalByAmount.sol
Comment (Pandapip1) [inline assets/eip-5216/ERC1155ApprovalByAmount.sol:147]:
```suggestion
}

```

File: assets/eip-5216/ERC1155ApprovalByAmount.sol
Comment (Pandapip1) [inline assets/eip-5216/ERC1155ApprovalByAmount.sol:34]:
```suggestion
```

File: assets/eip-5216/ERC1155ApprovalByAmount.sol
Comment (Pandapip1) [inline assets/eip-5216/ERC1155ApprovalByAmount.sol:4]:
```suggestion
import "IERC1155.sol";
import "ERC1155.sol";
```

File: assets/eip-5216/ERC1155ApprovalByAmount.sol
Comment (Pandapip1) [inline assets/eip-5216/ERC1155ApprovalByAmount.sol:1]:
```suggestion
// SPDX-License-Identifier: CC0-1.0
```

File: EIPS/eip-5216.md
Comment (ivanmmurciaua) [reply EIPS/eip-5216.md]: Changed

File: assets/eip-5216/ERC1155ApprovalByAmount.sol
Comment (ivanmmurciaua) [reply assets/eip-5216/ERC1155ApprovalByAmount.sol]: Added

File: assets/eip-5216/ERC1155ApprovalByAmount.sol
Comment (ivanmmurciaua) [reply assets/eip-5216/ERC1155ApprovalByAmount.sol]: Moved.

File: assets/eip-5216/ERC1155ApprovalByAmount.sol
Comment (ivanmmurciaua) [reply assets/eip-5216/ERC1155ApprovalByAmount.sol:4]: Done.

File: assets/eip-5216/ERC1155ApprovalByAmount.sol
Comment (ivanmmurciaua) [reply assets/eip-5216/ERC1155ApprovalByAmount.sol]: License changed. Thanks @Pandapip1

**Review by Pandapip1 — APPROVED** [review]
LGTM. Please close and re-open this PR so that it merges.

**Comment by ivanmmurciaua** [comment]
I don't know what is happening with eth-bot. @MicahZoltu @Pandapip1

**Comment by Pandapip1** [comment]
Oops it appears there might have been a bug.

**Review by lightclient — CHANGES_REQUESTED** [review]
Not sure why the bot is having issues.

**Comment by Pandapip1** [comment]
Oh. There's a problematic branch protection setting. Please disable the "require conversation resolution" option @lightclient.

**Comment by lightclient** [comment]
@Pandapip1 it isn't enabled.

**Pull request merged**
